plugins/SitesManager/API.php

```diff
@@ -1346,6 +1362,70 @@ public function setDefaultTimezone($defaultTimezone)
         return true;
     }
 
+    /**
+     * Sets global query parameter exclusion based on the specified exclusion type.
+     *
+     * @param string $exclusionType The type of query param exclusion, must be of the following:
+     *  - no_exclusions
+     *  - common_pii_exclusions
+     *  - custom_exclusions
+     * @param string|null $queryParamsToExclude (Optional) Comma separated list of query parameters to exclude when $exclusionType is 'custom'.
+     *                                         Ignored if $exclusionType is not 'custom'.
+     * @return void
+     * @throws Exception
+     */
+    public function setGlobalQueryParamExclusion(string $exclusionType, ?string $queryParamsToExclude = null): void
+    {
+        Piwik::checkUserHasSuperUserAccess();
+
+        $whiteListValidator = new WhitelistedValue(SitesManager::URL_PARAM_EXCLUSION_TYPES);
+        $whiteListValidator->validate($exclusionType);
+
+        if ($exclusionType === SitesManager::URL_PARAM_EXCLUSION_TYPE_NAME_CUSTOM_EXCLUSIONS && empty($queryParamsToExclude)) {
+            throw new Exception($this->translator->translate('SitesManager_ExceptionEmptyQueryParamsForCustomType'));
+        }
+
+        if ($exclusionType !== SitesManager::URL_PARAM_EXCLUSION_TYPE_NAME_CUSTOM_EXCLUSIONS && !empty($queryParamsToExclude)) {
+            throw new Exception($this->translator->translate('SitesManager_ExceptionNonEmptyQueryParamsForNonCustomType'));
+        }
+
+        Option::set(self::OPTION_EXCLUDE_TYPE_QUERY_PARAMS_GLOBAL, $exclusionType);
+
+        if ($exclusionType !== SitesManager::URL_PARAM_EXCLUSION_TYPE_NAME_CUSTOM_EXCLUSIONS) {
+            Option::delete(self::OPTION_EXCLUDED_QUERY_PARAMETERS_GLOBAL);
+            Cache::deleteTrackerCache();
+            return;
+        }
+
+        $excludedQueryParameters = $this->checkAndReturnCommaSeparatedStringList($queryParamsToExclude);
```

Comment:
It is currently possible to have the "custom" exclusion setting without any parameters set. This can be achieved by sending a non-empty blank string from the frontend.

And once you have achieved that state you need to change the exclusion type to be able to save the site manager settings again.

Maybe we should always run the parameter through the `checkAndReturn` function to prevent anyone from accidentally messing up their config? This could also prevent someone sending a list of commas from a badly behaving API client :thinking:

Reply:
Good point, I've updated this and added test coverage for it as well in the integration test.